consensus/ibft/bridge.go

```diff
@@ -140,6 +140,13 @@ func (b *BridgeMechanism) consumeStateTransactionsHook(numberParam interface{})
 		b.bridge.StateSync().Consume(tx)
 	}
 
+	// On every epoch end, create the checkpoint
+	if b.ibft.IsLastOfEpoch(block.Number()) {
+		if err := b.bridge.StartNewCheckpoint(b.ibft.epochSize); err != nil {
+			return err
+		}
+	}
```

Comment:
You can move this inner if check outside, and just return if the first check fails (is not the last of the epoch), to avoid having spaghetti

Reply:
Hmm, I am not sure I agree with you on this.  I agree that we should avoid spaghetti code but in my opinion, this way is clearer that checkpoints are created only at the end of the epoch
Also, everything below the first check will be unreachable if the condition is not met. I can imagine someone extending this method and adding something on the end of the method not thinking about the condition above. 
```
	if b.ibft.IsLastOfEpoch(block.Number()) {
		return nil
	}

	if err := b.bridge.StartNewCheckpoint(b.ibft.epochSize); err != nil {
		return err
	}

        .....
```

Reply:
Why do you think this
```go
        if b.ibft.IsLastOfEpoch(block.Number()) {
		if err := b.bridge.StartNewCheckpoint(b.ibft.epochSize); err != nil {
			return err
		}
	}
```

is cleaner than this?:
```go
        // ...
        if !b.ibft.IsLastOfEpoch(block.Number()) {
		return nil
	}

	return b.bridge.StartNewCheckpoint(b.ibft.epochSize)
```

Reply:
Fixed df8a837dd41fadcdf9bd23709ed0b2e78cd65aed